lib/vanagon/project/dsl.rb

```diff
@@ -171,6 +171,21 @@ def generate_archives(archive)
         @project.compiled_archive = archive
       end
 
+      # Sets the release for the project to the number of commits since the
+      # last tag. Requires that a git tag be present
+      # and reachable from the current commit in that repository.
+      #
+      def release_from_git
+        git_version = Git.open(File.expand_path("..", Vanagon::Driver.configdir)).describe('HEAD', tags: true)
+        # git describe actually contains the information we need here.
+        # for example: 0.15.2-5-g332dbcc, the -5- indicates there are 5
+        # commits since the last tag, so just split on '-' and take the
+        # second to last output
+        release(git_version.to_s.split('-')[-2])
+      rescue Git::GitExecuteError
+        warn "Directory '#{dirname}' cannot be versioned by git. Maybe it hasn't been tagged yet?"
+      end
```

Comment:
I dumped a longish comment and some suggested example code in https://github.com/puppetlabs/vanagon/pull/543#discussion_r166536715, and totally overlooked that you might not be getting notices or notice something buried in a comment attached to an outdated commit. Take a look at that for an alternate suggestion along the lines that led to the switch to `split`.

Reply:
@underscorgan do you have a particular preference on which approach to take?

I wouldn't mind adding the new git rev-list lib from Ryan's example: https://gist.github.com/mckern/89d2da3ad1ef7e1417e88424b3ac933f but I defer to you as to whether you want to add the additional complexity there.

Reply:
@McdonaldSeanp I like the approach Ryan had, can you give that a shot?

Reply:
Sure thing!

Reply:
(I know it's a lot more code than doing the `split` operation -- it won't break my heart if y'all opt not to extend the upstream Git library. I understand the tradeoffs on abstraction vs. maintainability vs. solving your specific problem that you actually have right now.)

Reply:
(I tossed a quick license on the example that basically gives you permission to use the code however you want to do whatever you want. Just dotting T's and crossing I's.)

![crosseyedemoji](https://user-images.githubusercontent.com/344926/36119097-dee013ae-0ff3-11e8-9680-32d1579e398a.jpg)